tasmota/tasmota_xsns_sensor/xsns_05_ds18x20.ino

```diff
@@ -63,8 +65,20 @@ struct {
 #ifdef DS18x20_USE_ID_ALIAS
   uint8_t alias;
 #endif //DS18x20_USE_ID_ALIAS  
+#ifdef DS18x20_MULTI_GPIOs
+  int8_t pins_id = 0; 
+#endif //DS18x20_MULTI_GPIOs
 } ds18x20_sensor[DS18X20_MAX_SENSORS];
 
+#ifdef DS18x20_MULTI_GPIOs
+struct {
+  int8_t pin = 0;            // Shelly GPIO3 input only
+  int8_t pin_out = 0;        // Shelly GPIO00 output only
+  bool dual_mode = false;    // Single pin mode
+} ds18x20_gpios[MAX_DSB];
+uint8_t ds18x20_ngpio = 0;    // Count of GPIO found
+#endif
+
```

Comment:
Why introducing a new data structure instead of modulating the size of the existing `DS18X20Data` between `[1]` and `[MAX_DSB]` 
That should simplify the rest of the code

Reply:
The new structure uses only 12 bytes of RAM. DS18X20Data uses 22 bytes and increasing to 4 copies will add another 66 bytes.
And will also have to pass GPIO numbers to all low-level functions: OneWire1ReadBit(void), OneWire2ReadBit(void), OneWireWriteBit(uint8_t v), OneWireReset(void). Such changes can lead to a significant increase in the final code.

Reply:
That's what I was wondering : passing some parameters to function vs copying

Reply:
I made a test module, replaced copying with transferring the index of the gpio array. 
With copying use 687300 bytes of flash.
With passing the parameter to the low-level functions 687364 bytes of flash. 
Loss of 64 bytes.

Reply:
Thanks for trying